### PR TITLE
[Java] Additional enum type annotations

### PR DESCRIPTION
**Pull request by borsch**
Original issue: https://github.com/OpenAPITools/openapi-generator/issues/6055
While original issue is about adding `additionalModelTypeAnnotations` to enum this is wrong solution since not all model annotations makes sence for enums. That's why I have added another option - `additonalEnumTypeAnnotations`

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by borsch** [comment]
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @nmuesch (2021/01)

File: bin/configs/spring-mvc.yaml [hunk 1]
```diff
@@ -10,3 +10,4 @@ additionalProperties:
   hideGenerationTimestamp: "true"
   serverPort: "8002"
   additionalModelTypeAnnotations: '@com.fasterxml.jackson.annotation.JsonFilter(value = "filter-name");@com.fasterxml.jackson.annotation.JsonIgnoreProperties(value = "id")'
+  additionalEnumTypeAnnotations: '@com.fasterxml.jackson.annotation.JsonTypeName("myName")'
```
Comment (wing328) [inline bin/configs/spring-mvc.yaml:13]: Seems to me the JsonTypeName is hardcoded to "myName" for all Enum models in the auto-generated code. What if the JsonTypeName is different for each Enum model?

Comment (borsch) [reply bin/configs/spring-mvc.yaml:13]: @wing328 this is just an example of usage/verify everything works. This feature is about adding common annotations to all enums

Comment (borsch) [reply bin/configs/spring-mvc.yaml]: @wing328 changed to `@com.fasterxml.jackson.annotation.JsonFormat`

**Review by Zomzog — APPROVED** [review]
LGTM

**Comment by borsch** [comment]
@Zomzog can this PR be merged?

**Comment by Zomzog** [comment]
> @Zomzog can this PR be merged?

I can't merge PRs.

@wing328 do you think this one can be merged?

**Comment by wing328** [comment]
I'll take a look tomorrow. Please put the restart on hold.

**Comment by wing328** [comment]
CircleCI tests passed via https://app.circleci.com/pipelines/github/OpenAPITools/openapi-generator/9309/workflows/971ad797-7217-4ea2-8653-4e898499bbe3/jobs/22902

**Pull request merged**